app/client/components/delivery/records/deliveryRecords.tsx

```diff
@@ -185,29 +182,23 @@ export const DeliveryRecordsFC = (props: DeliveryRecordsFCProps) => {
   const scrollToTop = () => window.scrollTo(0, 0);
   const resetDeliveryRecordsPage = () => setPageStatus(PageStatus.READ_ONLY);
 
-  const filterData = (productPartName: ProductUrlPart) => {
-    const NumOfRecordsToShow =
-      ProductTypes[productPartName as ProductTypeKeys].delivery?.records
-        ?.numberOfProblemRecordsToShow || props.data.results.length;
+  const productType = props.routeableStepProps.productType;
 
+  const filterData = () => {
     if (pageStatus !== PageStatus.READ_ONLY) {
+      const numOfReportableRecords =
+        productType.delivery.records.numberOfProblemRecordsToShow;
+
+      const today = moment();
+
+      const isNotHolidayProblem =
+        choosenDeliveryProblem !== holidaySuspensionDeliveryProblem.label;
+
       return props.data.results
```

Comment:
split this out into multiple `filters` which computationally is slightly less efficient but much easier to read